source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+/*
+   Copyright (C) 2015 by the authors of the ASPECT code.
+
+   This file is part of ASPECT.
+   ASPECT is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   ASPECT is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with ASPECT; see the file doc/COPYING.  If not see
+   <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/initial_conditions/interface.h>
+#include <aspect/initial_conditions/adiabatic_boundary.h>
+#include <fstream>
+#include <iostream>
+
+/**
+ * A class that computes an initial temperature field based
+ * on a user-defined adiabatic boundary for the 3D ellipsoid
+ * chunk geometry model. It discretizes the model domain into
+ * two regions separated by an isotherm boundary below which
+ * the temperature increases adiabatically. Above the
+ * user-defined isotherm boundary the temperature linearly
+ * increases from a surface temperature (273.15 K or 0 degree C)
+ * to the isotherm (1673.15 K or 1600 degree C). The user
+ * defines the location of the isotherm boundary with an ascii
+ * data file with the format defined in the ASPECT manual. Note
+ * that the latitudinal and longitudinal bounds of the ascii input
+ * data file need to be at least 1 degree wider than the bounds
+ * you use to define the region of your ellipsoid chunk geometry.
+ *
+ * This plugin is developed by Tahiry Rajaonarison, D. Sarah Stamps,
+ * and Wolfgang Bangerth.
```

Comment:
This documentation seems a duplicate of the one in the header file, which can lead to contradicting comments, if you ever change one of the two, but forget the other (in the best case you will remember and have additional effort to keep them synchronized). We usually do not include such a general description in the .cc file, and only keep the one in the header file, because that is the one users of your class will see.